### ISSUE TITLE
Problem loading font between frames

### ISSUE THREAD

**Issue by leonsal** [ocornut/imgui#5506]
(Click "Preview" above ^ to turn URL into clickable links)

1. FOR FIRST-TIME USERS ISSUES COMPILING/LINKING/RUNNING or LOADING FONTS,  please use [GitHub Discussions](https://github.com/ocornut/imgui/discussions).

2. PLEASE CAREFULLY READ: [FAQ](https://github.com/ocornut/imgui/blob/master/docs/FAQ.md)

3. PLEASE CAREFULLY READ: [Contributing Guidelines](https://github.com/ocornut/imgui/blob/master/docs/CONTRIBUTING.md)

4. PLEASE MAKE SURE that you have: read the FAQ; explored the contents of `ShowDemoWindow()` including the Examples menu; searched among Issues; used your IDE to search for keywords in all sources and text files; and read the links above.

5. Be mindful that messages are being sent to the e-mail box of "Watching" users. Try to proof-read your messages before sending them. Edits are not seen by those users.

6. Delete points 1-6 and PLEASE FILL THE TEMPLATE BELOW before submitting your issue.

Thank you!

----

_(you may also go to Demo>About Window, and click "Config/Build Information" to obtain a bunch of detailed information that you can paste here)_

**Version/Branch of Dear ImGui:**

Version: XXX
Branch: XXX _(master/viewport/docking/etc.)_

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_XXX.cpp + imgui_impl_XXX.cpp _(or specify if using a custom engine/back-end)_
Compiler: XXX _(if the question is related to building or platform specific features)_
Operating System: XXX

**My Issue/Question:**

XXX _(please provide as much context as possible)_

**Screenshots/Video**

XXX _(you can drag files here)_

**Standalone, minimal, complete and verifiable example:** _(see https://github.com/ocornut/imgui/issues/2261)_
```
// Here's some code anyone can copy and paste to reproduce your issue
ImGui::Begin("Example Bug");
MoreCodeToExplainMyIssue();
ImGui::End();
```

**Comment by leonsal** [comment]
**Version/Branch of Dear ImGui:**

Version: 1.89 WIP
Branch: master

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_glfw.cpp + imgui_impl_opengl3.cpp
Operating System: Linux Mint 20.3

**My Issue/Question:**

Trying to implement a splash screen while loading several fonts for my app.
The fonts are loaded after the first frames with the image are shown.
After the fonts are loaded the normal render loop is executed but the
windows are rendered completely black.
If the fonts aren't loaded between the frames it works OK with
the default font (see the following screenshots)

For testing I  am using a modified version of the `example_glfw_opengl3/main.cpp`.
No image is being loaded, just an empty frame before the normal render loop.

**Screenshots/Video**

Loading font between frames:
![Screenshot_problem](https://user-images.githubusercontent.com/7933396/180826066-8c504c33-34df-4b1c-b241-5f1c788704fa.png)

Commented out the loading of font between frames:
![Screenshot_ok](https://user-images.githubusercontent.com/7933396/180826084-d965a4c7-c191-44ec-ba14-d8c97af900f3.png)

Standalone, minimal, complete and verifiable example:
```
   // ...
   // PREVIOUS LINES ARE THE SAME FROM example_glfw_opengl3/main.cpp 

    // Our state
    bool show_demo_window = true;
    bool show_another_window = false;
    ImVec4 clear_color = ImVec4(0.45f, 0.55f, 0.60f, 1.00f);

    // Show first frames
    int count = 2;
    while (count > 0) {
        ImGui_ImplOpenGL3_NewFrame();
        ImGui_ImplGlfw_NewFrame();
        ImGui::NewFrame();

        // Splash screen image will be rendered here
        // ImGui::Image(...)
        
        ImGui::Render();
        int display_w, display_h;
        glfwGetFramebufferSize(window, &display_w, &display_h);
        glViewport(0, 0, display_w, display_h);
        glClearColor(clear_color.x * clear_color.w, clear_color.y * clear_color.w, clear_color.z * clear_color.w, clear_color.w);
        glClear(GL_COLOR_BUFFER_BIT);
        ImGui_ImplOpenGL3_RenderDrawData(ImGui::GetDrawData());
        glfwSwapBuffers(window);

        count--;
    }

    // Load fonts..
    io.Fonts->AddFontFromFileTTF("../../misc/fonts/Roboto-Medium.ttf", 28.0f);
    io.Fonts->Build();
    sleep(2); // simulates loading several fonts

    // Main loop
    while (!glfwWindowShouldClose(window))
    {
        // Poll and handle events (inputs, window resize, etc.)
        // You can read the io.WantCapture

   // NEXT LINES ARE THE SAME FROM example_glfw_opengl3/main.cpp 
   // ...
```

**Comment by PathogenDavid** [comment]
Thanks a lot for filling out the whole issue template!

You need to manually tell the backend to re-create the fonts texture, IE:

```cpp
// ...

// Load fonts..
io.Fonts->AddFontFromFileTTF("../../misc/fonts/Roboto-Medium.ttf", 28.0f);
io.Fonts->Build();
Sleep(2000); // simulates loading several fonts

ImGui_ImplOpenGL3_DestroyFontsTexture();
ImGui_ImplOpenGL3_CreateFontsTexture();

// Main loop
while (!glfwWindowShouldClose(window))
// ...
```

**Comment by leonsal** [comment]
It worked OK!
Thanks.

**Comment by ocornut** [comment]
FYI, work on `dynamic_fonts` branch aimed to be merged in 1.92 (#8465) solves this.
